pkg/controllers/rollout/controller.go

```diff
@@ -102,19 +103,27 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	//TODO: handle the case that a cluster was unselected by the scheduler and then selected again but the unselected binding is not deleted yet
 
-	// pick the bindings to be updated
-	tobeUpdatedBindings, needRoll := pickBindingsToRoll(allBindings, latestResourceSnapshotName, &crp)
+	// fill out all the default values for CRP just in case the mutation webhook is not enabled.
+	crpCopy := crp.DeepCopy()
+	fleetv1beta1.SetDefaultsClusterResourcePlacement(crpCopy)
+	// validate the clusterResourcePlacement just in case the validation webhook is not enabled
+	if err = validator.ValidateClusterResourcePlacement(crpCopy); err != nil {
+		klog.ErrorS(err, "Encountered an invalid clusterResourcePlacement", "clusterResourcePlacement", crpName)
+		return ctrl.Result{}, controller.NewUnexpectedBehaviorError(err)
+	}
+	// pick the bindings to be updated according to the rollout plan
+	toBeUpdatedBindings, needRoll := pickBindingsToRoll(allBindings, latestResourceSnapshotName, crpCopy)
 	if !needRoll {
 		klog.V(2).InfoS("No bindings are out of date, stop rolling", "clusterResourcePlacement", crpName)
 		return ctrl.Result{}, nil
 	}
-	klog.V(2).InfoS("Picked the bindings to be updated", "clusterResourcePlacement", crpName, "numberOfBindings", len(tobeUpdatedBindings))
+	klog.V(2).InfoS("Picked the bindings to be updated", "clusterResourcePlacement", crpName, "numberOfBindings", len(toBeUpdatedBindings))
 
-	// update all the bindings in parallel according to the rollout plan, requeue the request after the unavailable period
+	// update all the bindings in parallel according to the rollout plan, requeue the request after the mean value of unavailable period
 	// has passed so that some applied bindings will be considered "ready" by the rollout process regardless if the binding updates succeed or not.
 	// This is to avoid the case that the rollout process stalling because the time based binding readiness does not trigger any event.
-	return ctrl.Result{RequeueAfter: time.Duration(*crp.Spec.Strategy.RollingUpdate.UnavailablePeriodSeconds) * time.Second},
-		r.updateBindings(ctx, latestResourceSnapshotName, tobeUpdatedBindings)
+	return ctrl.Result{RequeueAfter: time.Duration(*crpCopy.Spec.Strategy.RollingUpdate.UnavailablePeriodSeconds/2) * time.Second},
```

Comment:
did not get why we need to divide in half.

Are we reserving some time for the controller to process the request from the queue?